### DISCUSSION TITLE
Does fetch/pull of a remote branch also download non-updated files

### DISCUSSION THREAD

**Discussion by Tang-70** [community/community#118683] [New to GitHub]
A question about download size: 

Does git fetch of a remote branch download a copy of the entire remote branch to your remote tracking branch (prior to a checkout/merge) or does it just download the updated files? Limited internet data allowance is an issue and the branch size is about 300MB with regular updates. I would ask the similar question about git pull. I am seeing conflicting indications on the internet.

If it downloads the whole branch then is there a way to seamlessly only download the updates?

**Comment by AbdrrahimDahmani** [comment]
Yes it download everything even files that haven't changed.
for your problem I guess you can use : `git config --unset core.filemode ` and `git config --global core.filemode false ` 

In Git, the git config core.filemode false command is used to set the core.filemode configuration option to false. This configuration option controls whether Git considers the execute permission bits of a file when determining whether the file has been modified. By setting this option to false, Git will ignore the execute permission bits of a file and only consider changes to the content of the file when determining whether it has been modified. This can be useful if you want to avoid downloading files that have only had their execute permission bits changed, but not their actual content. However, it's important to note that setting the core.filemode configuration option to false can also have some drawbacks. For example, if you make changes to the execute permission bits of a file and then set this option to false, Git will not consider those changes as modifications and will not track them. This can lead to confusion and make it difficult to maintain the correct execute permission bits for your files. It's generally recommended to only use this configuration option in specific cases where you need to avoid downloading files with only changed execute permission bits.

**Comment by jfullstackdev** [comment]
since it is a version control, it will know what files are only needed to be downloaded, and from Copilot : 

---

When you run `git fetch`, it does not download the entire branch each time. Instead, it downloads only the changes that have been made since the last time you fetched or pulled from that branch. This includes new commits, updated references, etc. The same applies to `git pull`, which is essentially a `git fetch` followed by a `git merge` or `git rebase`.

Git is designed to be efficient with data usage. It uses a packfile mechanism to compress and store objects in the repository. When you fetch or pull, Git fetches these packfiles, which contain only the changes, not the entire branch each time.

However, if the changes include large files or a lot of changes, the amount of data downloaded could still be significant. If you're working with large files, you might want to look into Git LFS (Large File Storage), which is an extension for versioning large files without storing them directly in the Git repository.

To further save data, you can fetch or pull less frequently, or use `git clone --depth 1` to create a shallow clone with only the latest snapshot of the code. But note that this last option has limitations, as it doesn't give you the full history of the project.

**Reply by Tang-70** [reply]
Thank you for your reply and also for the additional information.
